toolkits/confluence/arcade_confluence/tools/page.py

```diff
@@ -0,0 +1,236 @@
+    ],
+    title: Annotated[str, "The title of the page"],
+) -> Annotated[dict, "The page"]:
+    """Rename a page by changing its title."""
+    # Get the page to rename
+    client = ConfluenceClientV2(context.get_auth_token_or_empty())
+    page_id = await client.get_page_id(page_identifier)
+
+    page = await get_page(context, page_id, content_format=BodyFormat.STORAGE)
+    status = page.get("page", {}).get("status", "current")
+    content = page.get("page", {}).get("body", {}).get(BodyFormat.STORAGE, {}).get("value", "")
+    old_version_number = page.get("page", {}).get("version", {}).get("number", 0)
+
+    # Rename the page
+    payload = client.prepare_update_page_payload(
+        page_id=page_id,
+        status=status,
+        title=title,
+        body_representation=BodyFormat.STORAGE,
+        body_value=content,
+        version_number=old_version_number + 1,
+        version_message="Rename the page",
+    )
+    updated_page = await client.put(f"pages/{page_id}", json=payload)
+
+    return client.transform_page_response(updated_page)
+
+
+@tool(
+    requires_auth=Atlassian(
+        scopes=["read:page:confluence"],
+    )
+)
+async def get_page(
+    context: ToolContext,
+    page_identifier: Annotated[
+        str, "Can be a page's ID or title. Numerical titles are NOT supported."
+    ],
+) -> Annotated[dict, "The page"]:
+    """Retrieve a SINGLE page's content by its ID or title.
+
+    If a title is provided, then the first page with an exact matching title will be returned.
+
+    IMPORTANT: For retrieving MULTIPLE pages, use `get_pages_by_id` instead
+    for a massive performance and efficiency boost. If you call this function multiple times
+    instead of using `get_pages_by_id`, then the universe will explode.
+    """
+    client = ConfluenceClientV2(context.get_auth_token_or_empty())
+    if page_identifier.isdigit():
+        return await client.get_page_by_id(page_identifier, BodyFormat.STORAGE)
+    else:
+        return await client.get_page_by_title(page_identifier, BodyFormat.STORAGE)
+
+
+@tool(
+    requires_auth=Atlassian(
+        scopes=["read:page:confluence"],
+    )
+)
+async def get_pages_by_id(
+    context: ToolContext,
+    page_ids: Annotated[
+        list[str],
+        "The IDs of the pages to get. IDs are numeric. Titles of pages are NOT supported. "
+        "Maximum of 250 page ids supported.",
+    ],
+) -> Annotated[dict, "The pages"]:
+    """Get the content of MULTIPLE pages by their ID in a single efficient request.
+
+    IMPORTANT: Always use this function when you need to retrieve content from more than one page,
+    rather than making multiple separate calls to get_page, because this function is significantly
+    more efficient than calling get_page multiple times.
+    """
+    if not page_ids:
+        raise ToolExecutionError(message="The 'page_ids' parameter must be non-empty")
+    page_ids = page_ids[:250]
```

Comment:
As a general rule, I think it could be dangerous to do things like this without warning the tool caller. They'll get back a response thinking that every page_id was considered in the search, but not actually.

I tend to lean on being safe with:

```
if len(page_ids) > 250:
    raise RetryableToolError(f"page_ids must have up 250 items. Got {len(page_ids)}, instead.")
```